cardano-api/test/Test/Cardano/Api/Typed/Gen.hs

```diff
@@ -645,24 +645,36 @@ genMaybePraosNonce :: Gen (Maybe PraosNonce)
 genMaybePraosNonce =
   Gen.maybe (makePraosNonce <$> Gen.bytes (Range.linear 0 32))
 
-genProtocolParameters :: Gen ProtocolParameters
-genProtocolParameters =
-  ProtocolParameters
-    <$> ((,) <$> genNat <*> genNat)
-    <*> genRational
-    <*> genMaybePraosNonce
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genNat
-    <*> genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genLovelace
-    <*> genEpochNo
-    <*> genNat
-    <*> genRational
-    <*> genRational
-    <*> genRational
+genProtocolParameters :: ShelleyBasedEra era -> Gen (ProtocolParameters era)
+genProtocolParameters sbe =
+  case sbe of
+    ShelleyBasedEraShelley -> genPreAlonzo
+    ShelleyBasedEraAllegra -> genPreAlonzo
+    ShelleyBasedEraMary -> genPreAlonzo
+ where
+   genPreAlonzo =
+    ProtocolParameters
+      <$> ((,) <$> genNat <*> genNat)
+      <*> genRational
+      <*> genMaybePraosNonce
+      <*> genNat
+      <*> genNat
+      <*> genNat
+      <*> genNat
+      <*> genNat
+      <*> (Just <$> genLovelace)
+      <*> genLovelace
+      <*> genLovelace
+      <*> genLovelace
+      <*> genEpochNo
+      <*> genNat
+      <*> genRational
+      <*> genRational
+      <*> genRational
+      <*> return Nothing
+      <*> return Nothing
+      <*> return Nothing
+      <*> return Nothing
+      <*> return Nothing
+      <*> return Nothing
```

Comment:
Will these stay `Nothing` when we finish Alonzo, or is there a TODO item here?

Reply:
I should leave a TODO here